src/api-collector/main.go

```diff
@@ -153,48 +148,82 @@ func getAPISpecsUrls(ctx context.Context, client *github.Client, owner string, r
 	return m.OpenApiSpecs, nil
 }
 
-func downloadAPISpecs(repo string, specsUrls []string) []string {
-	var downloadedSpecs []string
-	for _, url := range specsUrls {
-		resp, err := http.Get(url)
-		if err != nil {
-			log.Printf("Error downloading API spec file: %v\n", err)
-			continue
-		}
-		defer resp.Body.Close()
-
-		if resp.StatusCode != http.StatusOK {
-			log.Printf("Bad HTTP status: %s\n", resp.Status)
-			continue
-		}
-		content, err := io.ReadAll(resp.Body)
+func downloadAPISpecs(ctx context.Context, client *github.Client, owner string, repo string) []string {
+	content, err := getAPISpecFromDir(ctx, client, owner, repo)
+	if err == nil {
+		specPath, err := saveAPISpec(content, repo)
 		if err != nil {
-			log.Printf("Error reading http response: %v\n", err)
-			continue
+			log.Fatalf("%v\n", err)
 		}
-		var spec OpenAPISpec
-		err = yaml.Unmarshal(content, &spec)
-		if err != nil {
-			log.Printf("Error parsing OpenAPI spec yaml format: %s\n", err)
-			continue
-		}
-		dirPath := path.Join("docs", repo, spec.Info.Version)
-		err = os.MkdirAll(dirPath, os.ModePerm)
+		return []string{specPath}
+	}
+	log.Printf("OpenAPI specs not found in the default location docs/api/openAPI.yaml. Proceeding with .tractusx metadata.")
+	var downloadedSpecs []string
+	specsUrls, err := getAPISpecsUrlsFromMetadata(ctx, client, owner, repo)
+	if err != nil {
+		log.Printf("%v\n", err)
+	}
+	for _, url := range specsUrls {
+		specContent, err := getAPISpecFromUrl(url)
 		if err != nil {
-			log.Printf("Error creating directory: %s\n", err)
+			log.Printf("%v\n",err)
 			continue
 		}
-		urlSplit := strings.Split(url, "/")
-		specName := urlSplit[len(urlSplit)-1]
-		filePath := path.Join(dirPath, specName)
-		err = os.WriteFile(filePath, content, 0644)
+		specPath, err := saveAPISpec(specContent, repo)
 		if err != nil {
-			log.Printf("Error saving OpenAPI spec content to file: %s\n", err)
+			log.Printf("%v\n",err)
 			continue
 		}
-		downloadedSpecs = append(downloadedSpecs, filePath)
-		log.Printf("OpenAPI spec %s downloaded successfully\n", specName)
+		downloadedSpecs = append(downloadedSpecs, specPath)
+		log.Printf("OpenAPI spec saved successfully\n", specPath)
 	}
 	return downloadedSpecs
 }
 
+func getAPISpecFromDir(ctx context.Context, client *github.Client, owner string, repo string) ([]byte, error) {
+    apiContent, _, _, err := client.Repositories.GetContents(ctx, owner, repo, API_SPEC_PATH, &github.RepositoryContentGetOptions{
```

Comment:
instead of defining this method, `getApiSpecFromUrl` could be used for this purpose passing this url as parameter:
`https://raw.githubusercontent.com/eclipse-tractusx/<project-name>/main/docs/api/openAPI.yaml`

Reply:
that's actually a great, simplifying hint, thanks! https://github.com/eclipse-tractusx/api-hub/pull/14/commits/343b66f1958b4d5ba49bc5007cb6d469f9a69e2b